src/pages/table/table-date-filter-forms.jsx

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+import React, { useEffect, useState } from 'react';
+import Calendar from '@cloudscape-design/components/calendar';
+import DateInput from '@cloudscape-design/components/date-input';
+import FormField from '@cloudscape-design/components/form-field';
+import TimeInput from '@cloudscape-design/components/time-input';
+
+export function DateTimeForm({ filter, operator, value, onChange }) {
+  // Using the most reasonable default time per operator.
+  const defaultTime = operator === '<' || operator === '>=' ? undefined : '23:59:59';
+  const [{ dateValue, timeValue }, setState] = useState(parseValue(value ?? '', defaultTime));
+
+  const onChangeDate = dateValue => {
+    setState(state => ({ ...state, dateValue }));
+  };
+
+  const onChangeTime = timeValue => {
+    setState(state => ({ ...state, timeValue }));
+  };
+
+  // Parse value from filter text when it changes.
+  useEffect(
+    () => {
+      filter && setState(parseDateTimeFilter(filter.trim()));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [filter]
+  );
+
+  // Call onChange only when the value is valid.
+  useEffect(
+    () => {
+      const dateAndTimeValue = dateValue + 'T' + (timeValue || '00:00:00');
+
+      if (!dateValue.trim()) {
+        onChange(null);
+      } else if (isValidIsoDate(dateAndTimeValue)) {
+        onChange(dateAndTimeValue);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [dateValue, timeValue]
+  );
+
+  return (
+    <div className="date-time-form">
+      <FormField description="Date" constraintText="Use YYYY/MM/DD format.">
+        <DateInput placeholder="YYYY/MM/DD" value={dateValue} onChange={event => onChangeDate(event.detail.value)} />
+      </FormField>
+
+      <FormField description="Time" constraintText="Use 24-hour format.">
+        <TimeInput
+          format="hh:mm:ss"
+          placeholder="hh:mm:ss"
+          value={timeValue}
+          onChange={event => onChangeTime(event.detail.value)}
+        />
+      </FormField>
+
+      <Calendar
+        value={dateValue}
+        locale="en-EN"
```

Comment:
not sure this is a valid locale, I think `en-GB` is UK English (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString#using_locales)